### ISSUE TITLE
Optimize BigInteger "DataTip" speed

### ISSUE THREAD

**Issue by GSPP** [dotnet/corefx#8967]
For very big BigIntegers the Visual Studio DataTip (the thing that appears when hovering over a variable with the mouse) takes a long time to load. I assume that's because the ToString processing takes too long.

The result is not visible anyway because it is aborted by a timeout as it seems.

Can the DataTip be made faster for super long BigIntegers? Maybe it could just show the first 100 digits instead of all digits. It's still possible to see all digits by calling ToString using the debugger.

**Comment by mellinoe** [comment]
An easy solution would be to add a `[DebuggerDisplayAttribute]` to BigInteger with the formatting you recommended. Right now it's just using `ToString`, like you mentioned, so it falls apart for very large values.

**Comment by GSPP** [comment]
Yes, I thought about `DebuggerDisplay`. Did not remember what it was called.
